Frameworks/AVFoundation/AVAudioPlayer.mm

```diff
@@ -160,27 +165,21 @@ - (instancetype)initWithData:(NSData*)data fileTypeHint:(NSString*)utiString err
                                          userInfo:nil];
         }
 
-        buffer.Attach(rawBuffer);
-
-        // WARNING: If someone deletes this AVAudioPlayer before the StoreAsync is completed, _data may be released
-        // causing IBuffer to segfault.
-
-        // TODO: subclassed IAsyncOperation<T>s don't get generated correctly in ObjCUWP yet, when that happens it'll
-        // open up StoreAsync.
-        ComPtr<IDataWriter> writer;
-        ComPtr<IAsyncOperation<UInt32>> comp;
-
-        [rw comObj].As(&writer);
+        winrt::attach_abi(buffer, reinterpret_cast<winrt::ABI::Windows::Storage::Streams::IBuffer*>(rawBuffer));
 
-        FAIL_FAST_HR_IF_NULL_MSG(E_UNEXPECTED, writer.Get(), "WSSDataWriter does not confrom to IDataWriter");
+        rw.WriteBuffer(buffer);
+        auto async = rw.StoreAsync();
 
-        writer->WriteBuffer(buffer.Get());
-        writer->StoreAsync(&comp);
-
-        [_mediaElement setSource:stream mimeType:@""];
-
-        // TODO: Is this needed?
-        [rw detachStream];
+        // Hook the stream up to the media control when the load is complete
+        async.Completed([self] (auto&& operation, auto&& status) {
+            if (status == winrt::Windows::Foundation::AsyncStatus::Completed) {
+                // Access XAML control on main thread
+                dispatch_sync(dispatch_get_main_queue(), ^{
+                    self->_mediaElement.SetSource(self->_stream, L"");
+                    self->_stream = nullptr;
+                });
+            }
+        });
```

Comment:
How did the old way ever work?
Did we just not call it?
Either way this looks correct.